src/main/java/org/recordrobotics/charger/subsystems/Arm.java

```diff
@@ -3,137 +3,157 @@
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
-public class Arm extends SubsystemBase {
-	private WPI_TalonFX _originMotor = new WPI_TalonFX(RobotMap.Arm.ORIGIN_MOTOR_PORT);
-	private WPI_TalonFX _changeMotor = new WPI_TalonFX(RobotMap.Arm.CHANGE_MOTOR_PORT);
-	private static final double FIRST_ARM_LENGTH = 30;
-	private static final double SECOND_ARM_LENGTH = 30;
-
-	// IN DEGREES
-	// negative is in the same direction as the rotation, positive is in the opposite
-	private static final double ORIGIN_START_OFFSET = -45;
-	private static final double CHANGE_START_OFFSET = 0;//-90;
 
+public class Arm extends SubsystemBase{
+    private WPI_TalonFX _originMotor = new WPI_TalonFX(RobotMap.Arm.ORIGIN_MOTOR_PORT);
+	private WPI_TalonFX _changeMotor = new WPI_TalonFX(RobotMap.Arm.CHANGE_MOTOR_PORT);
+	public static final double FIRST_ARM_LENGTH = Units.inchesToMeters(40); // TODO: All of these lengths and angles are off slightly, and should be modified
+	public static final double SECOND_ARM_LENGTH = Units.inchesToMeters(20); // TODO: check all of these constants
+    //private static final double FIRST_ARM_ZERO = Math.PI/3;
+    //private static final double SECOND_ARM_ZERO = Math.PI;
 	private static final double TICKS_PER_REV = 2048;
-	private static final double GEAR_RATIO = 16;
-	private static final double ERROR_MARGIN = 10;
+	private static final double GEAR_RATIO = 48;
+	// private static final double ERROR_MARGIN = 0;
+
+	private PIDController _originPid;
+	private PIDController _changePid;
+	private static final double C_KP = 0.005;
+	private static final double C_KI = 0.001;
+	private static final double C_KD = 0.0005;
+	private static final double O_KP = 0.025;
+	private static final double O_KI = 0;
+	private static final double O_KD = 0;
+	private double _changeTolerance = 0.5;
+	private double _originTolerance = 0.5;
+	private double _originMaxSpeed = 0.5;
+	private double _changeMaxSpeed = 0.5;
+	private double[] prevAngles = {0, 0};
 
 	private GenericEntry _entryAngles;
 
 	private static final TalonFXConfiguration armConfig = new TalonFXConfiguration();
 
 	private double[] _angles = new double[2];
-
-	public Arm() {
+    
+    public Arm() {
 		armConfig.initializationStrategy = SensorInitializationStrategy.BootToZero;
+		armConfig.primaryPID.selectedFeedbackSensor = FeedbackDevice.IntegratedSensor;
 		_originMotor.configAllSettings(armConfig);
 		_changeMotor.configAllSettings(armConfig);
-		//_originMotor.setSelectedSensorPosition(0 * ORIGIN_START_OFFSET * TICKS_PER_REV / 360 * GEAR_RATIO * Math.sqrt(10));
-		//_changeMotor.setSeglectedSensorPosition(CHANGE_START_OFFSET * TICKS_PER_REV / 360 * GEAR_RATIO * Math.sqrt(10));
 		_originMotor.setNeutralMode(NeutralMode.Brake);
 		_changeMotor.setNeutralMode(NeutralMode.Brake);
 		_originMotor.set(0);
 		_changeMotor.set(0);
 
+		_originPid = new PIDController(0, 0, 0);
+		_originPid.setD(O_KD);
+		_originPid.setI(O_KI);
+		_originPid.setP(O_KP);
```

Comment:
```suggestion
		_originPid = new PIDController(O_KP, O_KI, O_KD);
```
see https://docs.wpilib.org/en/stable/docs/software/advanced-controls/controllers/pidcontroller.html:
// Creates a PIDController with gains kP, kI, and kD
PIDController pid = new PIDController(kP, kI, kD);

make the same change with _changePID